doc/source/security/vault.rst

```diff
@@ -5,29 +5,210 @@ Information about HashiCorp Vault
+   
+.. _vault/custom-server.setting-up/starting:
+
+Starting the Vault server
+--------------------------------------------------------------------------------
+
+There are two options to start the Vault server. In dev mode, you can use it for
+experimentation: 
+
+.. code-block:: bash
+
+   $ vault server -dev
+
+Run this way, |vault| features automatic authentication and in-memory
+storage. |vault-server| is not suitable for production use in this mode.
+
+Alternatively, you can *deploy* |vault-server| in your environment. For
+deployment, you use *hcl* configuration files (custom format similar to JSON
+developed by |hashicorp-vault|) where you can set values of multiple
+|vault-server| options.
+
+.. code-block:: text
+
+   storage "consul" {
+   address = "127.0.0.1:8500"
+   path    = "vault/"
+   disable_mlock = true
+   }
+
+   listener "tcp" {
+   address     = "127.0.0.1:8200"
+   tls_disable = 1
+   disable_mlock = true
+   }
 
 .. seealso::
 
-  HashiCorp Documentation:
+   |hashicorp-vault| Documentation:
+      - `Deploying Vault server <https://learn.hashicorp.com/vault/getting-started/deploy>`_
+      - `Vault server configuration options <https://www.vaultproject.io/docs/configuration>`_
+
+To replace the *inmemory* storage used in *dev* mode, you may use |consul|, which
+provides a distributed key-value storage. |consul| is installed separately. The
+installation procedure for |consul| is similar to that for |vault|: download the
+installation package for your system and copy the |file.consul| binary to a
+directory in the ``PATH`` environment variable. Then, start |command.consul|:
+
+.. code-block:: bash
+
+   $ consul agent -dev
+
+Now, you can start |vault-server| passing the configuration file that you have
+prepared as the value of the `-config` parameter:
+
+.. code-block:: bash
 
-  Installing Vault
-  https://www.vaultproject.io/docs/install/index.html
+   $ vault server -config=my-conf.hcl
 
-  KV Secrets Engine - Version 1
-  https://www.vaultproject.io/docs/secrets/kv/kv-v1.html
+For each new storage backend, you need to initialize |vault-server|. The
+initialized storage is sealed, however.
 
-  Production Hardening
-  https://learn.hashicorp.com/vault/operations/production-hardening
+.. code-block:: bash
+
+   $ # Hint: Initialing a backend storage;
+   $ #       it will have to be unsealed at the next stage
+   $ vault operator init
+   Unseal Key 1: S6eWP5jRxKBHElx2oswxfkeZwwH/XWg5Ei1LaHt9GcK4
+   Unseal Key 2: Tnd+kJEeaO1yOIYTjFPgqSd8/dyUkprZw1zAYa62ISAs
+   Unseal Key 3: I/kdckz2ELaWSeQS/iL+EffsBnbmMVyoI+T1DrFX5ATM
+   Unseal Key 4: iQBHfQPtRqzPQYlBYpCtKeVWrE5C/OxB5O5jZDGYXEgz
+   Unseal Key 5: E5j8VdttAxC+b/6tARAJRybYfyS1AiDqq5D/wJ4QcEbo
+
+   Initial Root Token: s.oM1h6Inq6WOdppd5sML1gmpc
+   ...
+
+The storage is initialized but it is sealed. To be able to store encryption
+keys, the storage must be unsealed. The output of `vault operator init` command
```

Comment:
Explanation after the example.

Reply:
Unsure why we are not also including links to the Hashicorp documentation ?

Reply:
at this point, it is important to focus the attention on the output after vault init. Note the lead-in text for the example: "For each new storage backend, you need to initialize |vault-server|. The
initialized storage is sealed, however." So, the example follows the explanation. after the user runs the initialization command they get some output which is actually a link to the next step.

Reply:
@Oneiroi there links above this context and after it. Introduced by the "seealso" directive, such as the following:
.. seealso::
  |hashicorp-vault| Documentation:
      - `Deploying Vault server <https://learn.hashicorp.com/vault/getting-started/deploy>`_
      - `Vault server configuration options <https://www.vaultproject.io/docs/configuration>`_